Shared/Samples/Augment reality to navigate route/AugmentRealityToNavigateRouteView.swift

```diff
@@ -0,0 +1,253 @@
+            }
+        } else {
+            completion(polyline)
+        }
+    }
+    
+    /// Starts navigating the route.
+    @MainActor
+    private func startNavigation() async throws {
+        guard let routeResult else { return }
+        let routeTracker = RouteTracker(
+            routeResult: routeResult,
+            routeIndex: 0,
+            skipsCoincidentStops: true
+        )
+        guard let routeTracker else { return }
+        
+        routeTracker.voiceGuidanceUnitSystem = Locale.current.usesMetricSystem ? .metric : .imperial
+        
+        routeDataModel.routeTracker = routeTracker
+        
+        do {
+            try await routeDataModel.routeTask.load()
+        } catch {
+            throw error
+        }
+        
+        if routeDataModel.routeTask.info.supportsRerouting,
+           let reroutingParameters = ReroutingParameters(
+            routeTask: routeDataModel.routeTask,
+            routeParameters: routeDataModel.routeParameters
+           ) {
+            do {
+                try await routeTracker.enableRerouting(using: reroutingParameters)
+            } catch {
+                throw error
+            }
+        }
+        
+        statusText = "Navigation will start."
+        await startTracking()
+    }
+    
+    /// Starts monitoring multiple asynchronous streams of information.
+    private func startTracking() async {
+        await withTaskGroup(of: Void.self) { group in
+            group.addTask { await trackStatus() }
+            group.addTask { await trackVoiceGuidance() }
+        }
+    }
+    
+    /// Monitors the asynchronous stream of tracking statuses.
+    ///
+    /// When new statuses are delivered, update the route's traversed and remaining graphics.
+    private func trackStatus() async {
+        guard let routeTracker = routeDataModel.routeTracker else { return }
+        for await status in routeTracker.$trackingStatus {
+            guard let status else { continue }
+                switch status.destinationStatus {
+                case .notReached, .approaching:
+                    if let route = routeResult?.routes.first {
+                        let currentManeuver = route.directionManeuvers[status.currentManeuverIndex]
+                        statusText = currentManeuver.text
+                    }
+                case .reached:
+                    statusText = "You have arrived!"
+                @unknown default:
+                    break
+            }
+        }
+    }
+    
+    /// Monitors the asynchronous stream of voice guidances.
+    private func trackVoiceGuidance() async {
+        guard let routeTracker = routeDataModel.routeTracker else { return }
+        for try await voiceGuidance in routeTracker.voiceGuidances {
+            routeDataModel.speechSynthesizer.stopSpeaking(at: .word)
+            routeDataModel.speechSynthesizer.speak(AVSpeechUtterance(string: voiceGuidance.text))
+        }
+    }
```

Comment:
This method would make more sense on the model since it doesn't depend on anything on the view:

```swift
/// Monitors the asynchronous stream of voice guidances.
func trackVoiceGuidance() async {
    guard let routeTracker else { return }
    for try await voiceGuidance in routeTracker.voiceGuidances {
        speechSynthesizer.stopSpeaking(at: .word)
        speechSynthesizer.speak(AVSpeechUtterance(string: voiceGuidance.text))
    }
}
```